servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -568,4 +570,16 @@ public void sessionHeartbeat(long sessionId, List<Long> metrics) {
   public void updatePinList(Set<Long> pinnedInodes) {
     mBlockStore.updatePinnedInodes(pinnedInodes);
   }
+
+  /**
+   * Gets the file information.
+   *
+   * @param fileId the file id
+   * @return the file info
+   * @throws FileDoesNotExistException if the file does not exist
+   * @throws IOException if an I/O error occurs
+   */
+  public FileInfo getFileInfo(long fileId) throws FileDoesNotExistException, IOException {
```

Comment:
sounds good. In general we should keep block worker talking to block master only, unless we have very good reason to put file system master client here.